crates/control/src/calibration_controller.rs

```diff
@@ -0,0 +1,277 @@
+use std::{time::Duration, vec};
+
+use color_eyre::Result;
+use serde::{Deserialize, Serialize};
+
+use calibration::{corrections::Corrections, measurement::Measurement, solve};
+use context_attribute::context;
+use coordinate_systems::Ground;
+use framework::{AdditionalOutput, MainOutput, PerceptionInput};
+use linear_algebra::{point, Point2};
+use types::{
+    calibration::{CalibrationCaptureResponse, CalibrationCommand},
+    camera_position::CameraPosition,
+    cycle_time::CycleTime,
+    field_dimensions::FieldDimensions,
+    primary_state::PrimaryState,
+};
+
+#[derive(Deserialize, Serialize)]
+pub struct CalibrationController {
+    current_look_at_index: usize,
+    current_calibration_state: CalibrationState,
+    current_measurements: Vec<Measurement>,
```

Comment:
I'd remove the prefix of these fields.
They represent the state and there is no need to differentiate them from past or futures states.

Reply:
Yeah, makes more sense. Fixed.